w09h03/test/pgdp/filter/DecodeTest.java

```diff
@@ -0,0 +1,44 @@
+package pgdp.filter;
+
+import org.bytedeco.javacv.FrameGrabber;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+public class DecodeTest {
+
+  private void closeProvider(FrameProvider fp) {
+    try {
+      fp.close();
+    } catch (FrameGrabber.Exception e) {
+      Assertions.fail(e);
+    }
+  }
+
+  @Test
+  @DisplayName("Decode first frame of noot")
+  public void testDecode_firstFrame() {
+    // read the file
+    VideoContainer in;
+    FrameProvider fp = new FrameProvider("noot.mp4");
+
+    Frame first = null;
+    try {
+      in = new VideoContainer(fp);
```

Comment:
```suggestion
```

do not instantiate the unused VideoContainer since this can lead to the first frame getting skipped

Reply:
Fixed